plugins/main/macvlan/macvlan.go

```diff
@@ -107,12 +107,35 @@ func loadConf(args *skel.CmdArgs, envArgs string) (*NetConf, string, error) {
 	if err := json.Unmarshal(args.StdinData, n); err != nil {
 		return nil, "", fmt.Errorf("failed to load netconf: %v", err)
 	}
-	if n.Master == "" {
-		defaultRouteInterface, err := getNamespacedDefaultRouteInterfaceName(args.Netns, n.LinkContNs)
+
+	var result *current.Result
+	var err error
+	// Parse previous result
+	if n.NetConf.RawPrevResult != nil {
+		if err = version.ParsePrevResult(&n.NetConf); err != nil {
+			return nil, "", fmt.Errorf("could not parse prevResult: %v", err)
+		}
+
+		result, err = current.NewResultFromResult(n.PrevResult)
 		if err != nil {
-			return nil, "", err
+			return nil, "", fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+	if n.Master == "" {
+		if result == nil {
+			var defaultRouteInterface string
+			defaultRouteInterface, err = getNamespacedDefaultRouteInterfaceName(args.Netns, n.LinkContNs)
+			if err != nil {
+				return nil, "", err
+			}
+			n.Master = defaultRouteInterface
+		} else {
+			if len(result.Interfaces) == 1 && result.Interfaces[0].Name != "" {
+				n.Master = result.Interfaces[0].Name
+			} else {
+				return nil, "", fmt.Errorf("chained master failure. PrevResult lacks a single named interface")
```

Comment:
I am thinking that we could provide a way to pick interface from prevResult by 'master' option.
I'm just wondering how about to pick interface by 'master' option name if prevResult != nil and multiple interfaces in result.

Reply:
`master` option is a static option in cni config.  how to dynamic config for different master? Give a special master option value to tell maclvan plugin to get master from prevResult ?